postgres/src/rows.rs

```diff
@@ -259,13 +256,21 @@ impl<'a> Row<'a> {
 pub trait RowIndex {
     /// Returns the index of the appropriate column, or `None` if no such
     /// column exists.
-    fn idx(&self, stmt: &Statement) -> Option<usize>;
+    fn idx<I: AsRef<StatementInfo>>(&self, info: &I) -> Option<usize> {
```

Comment:
Since we're changing it, maybe make it just take a `&[Column]` to avoid making `StatementInfo` public?